sorc/build_gsi.sh

```diff
@@ -30,6 +30,9 @@ fi
 cd gsi.fd/ush/
 export GSI_MODE="GFS"
 export REGRESSION_TESTS="NO"
+# Temporary setting until GSI default is updated or workflow adds versions file
```

Comment:
add @MichaelLueken-NOAA for awareness

Reply:
@MichaelLueken-NOAA 
`ncio/1.1.2` is now available on WCOSS2 in the para space.
Here are the instructions to find them:
```
module purge;module reset
export HPC_OPT=/apps/ops/para/libs
module use /apps/ops/para/libs/modulefiles/compiler/intel/19.1.3.304/
module use /apps/ops/para/libs/modulefiles/mpi/intel/19.1.3.304/cray-mpich/8.1.7/
module load intel craype cray-mpich/8.1.7
module show ncio/1.1.2
```

We will need to find a way to ensure development continues without having to depend on updates to libraries to fall in their final place on WCOSS2.   Does that mean test them on Acorn where we have control over building and installing versions of updated libraries?
@arunchawla-NOAA @dtkleist

Reply:
@aerorahul
I added:

```
pushenv("HPC_OPT", "/apps/ops/para/libs")
prepend_path("MODULEPATH", "/apps/ops/para/libs/modulefiles/compiler/intel/19.1.3.304")
prepend_path("MODULEPATH", "/apps/ops/para/libs/modulefiles/mpi/intel/19.1.3.304/cray-mpich/8.1.7/")
```

to `modulefiles/gsi_wcoss2.lua` and attempted to compile.  While `ncio/1.1.2` was found, the code failed to compile with the following:

```
CMake Error at src/gsi/CMakeLists.txt:68 (find_package):
  By not providing "Findnemsio.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "nemsio", but
  CMake did not find one.
```

This is strange because `nemsio/2.5.2` was also found.  Also, the following printout shows that the modules were being loaded from para:

```
-- Found bacio: /apps/ops/para/libs/intel/19.1.3.304/bacio/2.4.1/lib/libbacio_4.a (found version "2.4.0")
-- Found sigio: /apps/ops/para/libs/intel/19.1.3.304/sigio/2.3.2/lib/libsigio.a (found version "2.3.0")
-- Found sfcio: /apps/ops/para/libs/intel/19.1.3.304/sfcio/1.4.1/lib/libsfcio.a (found version "1.4.0")
```

rather than:

```
-- Found bacio: /apps/ops/prod/libs/intel/19.1.3.304/bacio/2.4.1/lib/libbacio_4.a (found version "2.4.0")
-- Found sigio: /apps/ops/prod/libs/intel/19.1.3.304/sigio/2.3.2/lib/libsigio.a (found version "2.3.0")
-- Found sfcio: /apps/ops/prod/libs/intel/19.1.3.304/sfcio/1.4.1/lib/libsfcio.a (found version "1.4.0")
```

So, while `ncio/1.1.2` is available in para, the GSI won't compile when the modulefile has been updated to use modules from para.

While I certainly agree that development continues without needing to depend on updates to libraries to fall in their final place on WCOSS2, at least with respect to `ncio/1.1.2`, it seems that it will need to.

Reply:
@MichaelLueken-NOAA 
This worked for me on Cactus:
```
12:53 $ git diff
diff --git i/modulefiles/gsi_wcoss2.lua w/modulefiles/gsi_wcoss2.lua
index 13894625..b97bbba7 100644
--- i/modulefiles/gsi_wcoss2.lua
+++ w/modulefiles/gsi_wcoss2.lua
@@ -19,5 +19,11 @@ load(pathJoin("python", python_ver))
 load(pathJoin("prod_util", prod_util_ver))
 
 load("gsi_common")
+unload("ncio")
+
+pushenv("HPC_OPT", "/apps/ops/para/libs")
+prepend_path("MODULEPATH", "/apps/ops/para/libs/modulefiles/compiler/intel/19.1.3.304")
+prepend_path("MODULEPATH", "/apps/ops/para/libs/modulefiles/mpi/intel/19.1.3.304/cray-mpich/8.1.7/")
+load("ncio/1.1.2")
 
 whatis("Description: GSI environment on WCOSS2")
```

It built everything as expected.

Reply:
I'm not really worried about it right now. I know the GSI update is being held up by WCOSS2, but I have a work-around and develop of GW doesn't run on WCOSS2 right now anyway. Updating GW to run on WCOSS2 will include the introduction of the versions file, so we'll be setting the ncio version number then anyway.

Reply:
> @MichaelLueken-NOAA This worked for me on Cactus at `/lfs/h2/emc/stmp/rahul.mahajan/gsi`
> 
> ```
> 12:53 $ git diff
> diff --git i/modulefiles/gsi_wcoss2.lua w/modulefiles/gsi_wcoss2.lua
> index 13894625..b97bbba7 100644
> --- i/modulefiles/gsi_wcoss2.lua
> +++ w/modulefiles/gsi_wcoss2.lua
> @@ -19,5 +19,11 @@ load(pathJoin("python", python_ver))
>  load(pathJoin("prod_util", prod_util_ver))
>  
>  load("gsi_common")
> +unload("ncio")
> +
> +pushenv("HPC_OPT", "/apps/ops/para/libs")
> +prepend_path("MODULEPATH", "/apps/ops/para/libs/modulefiles/compiler/intel/19.1.3.304")
> +prepend_path("MODULEPATH", "/apps/ops/para/libs/modulefiles/mpi/intel/19.1.3.304/cray-mpich/8.1.7/")
> +load("ncio/1.1.2")
>  
>  whatis("Description: GSI environment on WCOSS2")
> ```
> 
> It built everything as expected.

@aerorahul Thank you for looking into this.  The GSI was able to compile following what you have done.  I'll move forward with updating the GSI modulefiles to use ncio/1.1.2 now.